File: Server/PowerCraft/Tools/Writer.vb
Imports System.Collections.Generic
Imports System.IO
Imports System.Linq
Imports System.Text

Namespace PowerCraft.Tools

    Public Enum LogType
        Normal
        Warning
        Serious
    End Enum
    Public Class WriterAdapter
        Private Function Time() As [String]
            Return "<" + DateTime.Now.ToString("h:mm:ss tt") + ">"
        End Function

    End Class
    Public Class Writer
        Private text As [String]
        Public Sub New(textTobeWritten As [String])
            text = textTobeWritten
        End Sub
        Public Sub WriteToTextFile(filePath As [String])
            Dim sWriter As StreamWriter = File.CreateText(filePath)
            sWriter.Write(text)
            sWriter.Close()
        End Sub
        Public Shared wAdapter As WriterAdapter = Nothing
        Public Sub WriteToConsole(lType As LogType)
            If text Is Nothing Then
                Throw New ArgumentNullException("message")
            End If
            If text.Contains(ControlChars.Lf) Then
                For Each line As String In text.Split(ControlChars.Lf)
                    Dim w As New Writer(line)
                    w.WriteToConsole(lType)
                Next
                Return
            End If
            Dim processedMessage As String = "# "
            For i As Integer = 0 To text.Length - 1
                If text(i) = "&"c Then
                    i += 1
                Else
                    processedMessage += text(i)
                End If
            Next
            Log(lType, processedMessage)
        End Sub
        Public Sub Log(l As LogType, message As [String])

        End Sub
    End Class
End Namespace

File: PowerCraft/Tools/Writer.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;

namespace PowerCraft.Tools
{
    
    public enum LogType
    {
        Normal,
        Warning,
        Serious
    }
    public class WriterAdapter
    {
        private static String Time()
        {
            return "<" + DateTime.Now.ToString("h:mm:ss tt") + ">";
        }
        public static String PreviousLine = null;
        public static String MostRecentLine = "---------";
        public static int lineCount = 0;
        public static String[] lines = new String[lineCount];
        public static void AddTo(String text)
        {
            lineCount++;
            String[] OldArray = lines;
            String[] NewArray = new String[lineCount + 1];
            for (int lin = 0; lin < lineCount; lin++)
            {
                try
                {
                    NewArray[lin] = OldArray[lin];
                }
                catch (IndexOutOfRangeException ex)
                {

                }
            }
            PreviousLine = MostRecentLine;
            NewArray[lineCount] = "\n" + Time() + " " + text;
            MostRecentLine = NewArray[lineCount];
        }
    }
    public class Writer
    {
        private String text;
        public Writer(String textTobeWritten)
        {
            text = textTobeWritten;
        }
        public void WriteToTextFile(String filePath)
        {
            if (File.Exists(filePath)) File.Delete(filePath);
            StreamWriter sWriter = File.CreateText(filePath);
            sWriter.Write(text);
        }
        public void WriteToConsole(LogType lType)
        {
            WriterAdapter.AddTo(text);
        }
    }
}
